Compilation fails on OS X due to `getenv` not being a member of namespace `std`

Compilation errors occured in two Travis CI build jobs when spdlog commit def86e6 (HEAD of origin/master as of writing this) was checked out. Xcode images version 7.3 and 6.4 were used, with Apple Clang/LLVM version 7.3.0 and 6.1.0 as compilers, respectively. I can't verify this issue myself, as I don't own an OS X machine.

The error message:
```
.../spdlog/include/spdlog/details/os.h:436:25: error: no member named 'getenv' in namespace 'std'; did you mean simply 'getenv'?
    const char *env_p = std::getenv("TERM");
                        ^~~~~
/usr/include/stdlib.h:144:7: note: 'getenv' declared here
    char *getenv(const char *);
```

We can see the compiler being unable to find `std::getenv`. The function is declared in the headers `<stdlib.h>` (without `std::`) and `<cstdlib>` (with `std::`). The probable cause for this is, that `<cstdlib>` isn't included anywhere, but `<stdlib.h>` is.

At this moment, the includes of `details/os.h` are:

```
#include <spdlog/common.h>

#include <cstdio>
#include <ctime>
#include <functional>
#include <string>
#include <chrono>
#include <thread>
#include <algorithm>
#include <stdio.h>
#include <string.h>
#include <sys/stat.h>
#include <sys/types.h>
```

We can see a mixture of C (`<stdio.h>` and `<string.h>`) and C++ (`<cstdio>` and `<ctime>`) implementations of the C standard library headers being included. I'd propose the removal of `<stdio.h>`, changing `<string.h>` to `<cstring>` and addition of `<cstdlib>` to the included headers of `detail/os.h`.

This error doesn't occur on my system (Arch Linux, libstdc++ version 3.4), probably because the headers are different and include `<cstdlib>` somewhere.

The specific build jobs with logs are here:
https://travis-ci.org/varuna-lang/varuna/jobs/233227104 (Xcode 7.3, Apple clang 7.3.0)
https://travis-ci.org/varuna-lang/varuna/jobs/233227106 (Xcode 6.4, Apple clang 6.1.0)

I'll submit a PR if it's needed.

--- a/include/spdlog/details/os.h
+++ b/include/spdlog/details/os.h
@@ -13,8 +13,8 @@
 #include <chrono>
 #include <thread>
 #include <algorithm>
-#include <stdio.h>
-#include <string.h>
+#include <cstring>
+#include <cstdlib>
 #include <sys/stat.h>
 #include <sys/types.h>
 

